tensorflow/core/common_runtime/dml/dml_adapter_impl.cc

```diff
@@ -169,46 +169,72 @@ uint64_t DmlAdapterImpl::QueryAvailableLocalMemory() const {
 }
 
 std::vector<DmlAdapterImpl> EnumerateAdapterImpls() {
-  ComPtr<IDXGIFactory6> dxgi_factory;
-  DML_CHECK_SUCCEEDED(CreateDXGIFactory(IID_PPV_ARGS(&dxgi_factory)));
+  ComPtr<IDXGIFactory4> dxgi_factory;
+  HRESULT hr = CreateDXGIFactory(IID_PPV_ARGS(&dxgi_factory));
+  if (FAILED(hr)) {
+    LOG(WARNING) << "CreateDXGIFactory failed with HRESULT " << hr;
+    return {};
+  }
 
-  const D3D_FEATURE_LEVEL min_feature_level = D3D_FEATURE_LEVEL_11_0;
   std::vector<DmlAdapterImpl> adapter_infos;
 
-  uint32_t adapter_index = 0;
-  ComPtr<IDXGIAdapter1> adapter;
-  while (dxgi_factory->EnumAdapterByGpuPreference(
-             adapter_index, DXGI_GPU_PREFERENCE_HIGH_PERFORMANCE,
-             IID_PPV_ARGS(&adapter)) != DXGI_ERROR_NOT_FOUND) {
+  auto IsSoftwareAdapter = [](IDXGIAdapter1* adapter) -> bool {
```

Comment:
This seems like it'd be better off just being its own function, rather than being stuffed into EnumerateAdapterImpls.